nix/tullia.nix

```diff
@@ -0,0 +1,72 @@
+self: system:
+
+let
+  ciInputName = "GitHub event";
+in rec {
+  tasks = let
+    inherit (self.inputs.tullia) flakeOutputTasks taskSequence;
+
+    common = {
+      config,
+      ...
+    }: {
+      preset = {
+        # needed on top-level task to set runtime options
+        nix.enable = true;
+
+        github-ci = {
+          enable = config.actionRun.facts != {};
+          repo = "input-output-hk/plutus";
+          sha = config.preset.github-ci.lib.getRevision ciInputName null;
+        };
+      };
+    };
+
+    # the attribute name in `hydraJobs` for the current system
+    os = {
+      x86_64-linux = "linux";
+      x86_64-darwin = "macos";
+    }.${system};
+
+    mkHydraJobTask = flakeOutputTask: {
+      imports = [common flakeOutputTask];
+
+      # some hydra jobs run NixOS tests
+      env.NIX_CONFIG = ''
+        extra-system-features = kvm
+      '';
+
+      memory = 1024 * 8;
+      nomad.resources.cpu = 10000;
+    };
+    mkHydraJobTasks = __mapAttrs (_: mkHydraJobTask);
+
+    hydraJobTasks = mkHydraJobTasks (flakeOutputTasks ["hydraJobs" os] self);
+
+    ciTasks = taskSequence "ci/" hydraJobTasks (__attrNames hydraJobTasks);
+  in
+    ciTasks // {
+      "ci" = {lib, ...}: {
+        imports = [common];
+        after = [(lib.last (__attrNames ciTasks))];
```

Comment:
background question: is this a graph-like dependency ordering? i.e. could we say `after = _attrNames ciTasks` to say that it has to run after all the other tasks, and then rely on `taskSequence` to ensure that all the others run sequentially? Or do we actually have to say it runs specifically after the last one?

Reply:
Right, that should work too. I'll give it a try